doc/developer/design/20220314_linearized_reads.md

```diff
@@ -0,0 +1,160 @@
+
+The behavior of non-`CONSISTENT` reads does not need to be changed.
+They still consult `since` for their inputs, and may elect to choose any timestamp that is greater than this, but perhaps one that is not as "current" as for linearizable reads, to be able to return immediately.
+
+This behavior applies equally well to matarialized and unmaterialized sources.
+
+### Other open questions
+
+- How much of a latency hit will fetching the max offset incur?
+- How feasible is it for ALL sources to offer a "max offset" API?
+  - Can we degrade the available options for sources that offer no "max offset"?
+  - Will we require a heartbeat mechanism for those sources?
+- Will linearized reads negatively impact sequentially consistent reads (are they forced to advance similarly)?
+- What syntax is best to express linearizable (or non) reads? For example, as part of `BEGIN`?
+- Should be eventually change the default experience of using `materialized` to be `CONSISTENT`?
+
+## Formalism and constraints
+
+**Note**
+Please refer to [formalism](https://github.com/MaterializeInc/materialize/blob/fe0f4bac40db1afd21606428be14197601bb5a1c/doc/developer/platform/formalism.md)
+for more background about some of the terms used here, as well as
+this [this jepsen map on consistency](https://jepsen.io/consistency) for details
+about specific consistency models.
+
+Materialized offers reads with stronger _real-time_ consistency guarantees than the default.
+
+Reads again some view `V` always happen at a singular `timestamp`, which must be bound
+by the `upper` and `since` of that view. That view `V` is composed of many such _source_
+pTVC's, which are pTVC's that correspond to the data produced some some _external system_.
+
+True _linearizability_ requires that reads and writes happen in some _total order_ that is
+consistent with the _real time ordering_ of those operations. That is, each operation *appears*
+to happen atomically at some singular point in time bounded by the beginning and end of that
+operation.
+
+A `LINEARIZABLE` read against some view `V` is possible if-and-only-if the following are true:
+
+* All _source_ pTVC's that are at the base of `V` offer a _linearizable_ API for obtaining the
+  latest offset of data for that pTVC.
+  * "offset" here refers to some monotonically increasing identifier that durably and consistently
+  labels data.
+  * A given API is _linearizable_ if-and-only-if later (as in, later in _real-time_) calls tho
+  that API can never produce lower "offets".
+* For all _source_ pTVC's involved, materialized durably associate _real-time_ timestamps
+with specific offsets (these are typically referred to as _timestamp bindings_ and are currently
+implemented for Kafka sources).
+* ...and the responses from those API's can be given some _real-time total order_.
+  * The simplest way to obtain this constraint is to involve only one source.
```

Comment:
I feel like this sentence should either be removed or expanded. Why is it not simple to involve more than one source and make requests to each of them?

Reply:
It is not simple to get linearizability across those sources, but you do get linearizability for the data within each source...i think...I am going to discuss this further over the next few days